### DISCUSSION TITLE
How to use NPM workspaces when different packages use different major versions

### DISCUSSION THREAD

**Discussion by mesqueeb** [npm/feedback#236] [CLI] [answered]
I have a NPM7 monorepo that holds a core lib focussed on data communication with a plugin system.
Some plugin in my workspace in `./packages` use an NPM package (let's call it Y) of major version 2, while another uses Y of major version 3.

When the "docs" SPA in `./packages` uses Y at v2 as well as the plugin that uses Y at v2, the docs cannot be launched because it's giving a conflict for version of package Y. It tries to use Y at v3.

How can I prevent "Y" from being hoisted in the NPM7 workspace? It's the only issue.

**Comment by MylesBorins** [comment, marked as answer]
Hey @mesqueeb would you be able to open this as an issue over on https://github.com/npm/cli

This discussions form is generally about requesting new features in npm products, not for support.

Thanks!

**Reply by trusktr** [reply]
There does not appear to be an avenue for posting such a question at https://github.com/npm/cli
